src/components/MessageInput/hooks/useMessageInputState.ts

```diff
@@ -128,16 +130,19 @@ export type MessageInputHookProps<
   onPaste: (event: React.ClipboardEvent<HTMLTextAreaElement>) => void;
   onSelectUser: (item: UserResponse<StreamChatGenerics>) => void;
   recordingController: RecordingController;
-  removeAttachment: (id: string) => void;
+  removeAttachments: (ids: string[]) => void;
   removeFile: (id: string) => void;
   removeImage: (id: string) => void;
   textareaRef: React.MutableRefObject<HTMLTextAreaElement | null | undefined>;
   uploadAttachment: (
-    attachment: LocalAttachment<StreamChatGenerics>,
-  ) => Promise<LocalAttachment<StreamChatGenerics>>;
+    attachment: AnyLocalAttachment<StreamChatGenerics>,
+  ) => Promise<AnyLocalAttachment<StreamChatGenerics> | undefined>;
   uploadFile: (id: string) => void;
   uploadImage: (id: string) => void;
   uploadNewFiles: (files: FileList | File[]) => void;
+  upsertAttachments: (
+    attachments: (Attachment<StreamChatGenerics> | LocalAttachment<StreamChatGenerics>)[],
+  ) => void;
```

Comment:
```suggestion
  upsertAttachments: (
    attachments: (Attachment<StreamChatGenerics> | LocalAttachment<StreamChatGenerics>)[],
  ) => void;
```
nit/feat: Could integrators benefit from patch function?

```ts
upsertAttachments((currentAttachments) => {
  return currentAttachments.map((attachment) => {
    if (attachment.id !== arbitraryId) return attachment;
    return { ...attachment, localMetadata: { ...attachment.localMetadata, customThing: 123 } };
  });
});
```